src/infomgmag/mars/CountryAgent.java

```diff
@@ -84,9 +97,59 @@ public boolean bordersEnemy(){  //checks if a territory borders an enemy
         return bordersenemies;
     }
 
-    public double ownWholeContinent() {
+    public boolean ownWholeContinent(){     //checks if the agents owns the whole continent except this territory
+        int territoriesOwned = 0;
+        for (Territory ter : territory.getBelongsTo().getTerritories()){
+            if (ter.getOwner() == territory.getOwner()){
+                territoriesOwned += 1;
+            }
+        }
+        if (territoriesOwned == territory.getBelongsTo().getTerritories().size() - 1){
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public boolean enemyOwnsAnEntireContinent(){
+        int territoriesOwned = 0;
+        for (Territory ter : territory.getBelongsTo().getTerritories()){
+            if (ter.getOwner() == territory.getOwner()){
+                territoriesOwned += 1;
+            }
+        }
+        if (territoriesOwned == territory.getBelongsTo().getTerritories().size()){
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public int numberOfContinentsBordered(){       //checks the amount of continents bordered that are not the territories 'home' continent
```

Comment:
Perhaps this would be better as a territory attribute (and then you could set it in the Board constructor instead of recomputing it all the time). 

Instead of checking for uniqueness yourself, why not use the Set class? Also could be done with streams imo but I can understand if that's too much.